west.yml

```diff
@@ -79,6 +79,11 @@ manifest:
       path: modules/hal/gigadevice
       groups:
         - hal
+    - name: hal_hpmicro
+      revision: 83806f293cf2709e84f4649423c9729b2f507766
```

Comment:
this module seems to include unnecessary stuff:

https://github.com/zephyrproject-rtos/hal_hpmicro/blob/zephyr/hpm_sdk/arch/riscv/riscv_core.h#L17, see include/zephyr/arch/riscv/csr.h
https://github.com/zephyrproject-rtos/hal_hpmicro/blob/zephyr/hpm_sdk/cmake/cmake-ext.cmake, why is this needed?
https://github.com/zephyrproject-rtos/hal_hpmicro/blob/zephyr/hpm_sdk/soc/HPM6750/CMakeLists.txt

Please, make sure to only include stuff used in Zephyr context.

Reply:
`include/zephyr/arch/riscv/csr.h` will be called when the interrupt function is processed.

https://github.com/zephyrproject-rtos/hal_hpmicro/blob/zephyr/hpm_sdk/cmake/cmake-ext.cmake because https://github.com/hpmicro/hpm_sdk 's cmake uses these definitions, hpm_sdk is updated once a quarter, so that we can quickly apply updates to zephyr_hal when hpm_sdk is updated. This approach should not pose any problems.

https://github.com/zephyrproject-rtos/hal_hpmicro/blob/zephyr/hpm_sdk/soc/HPM6750/CMakeLists.txt  This issue is confirmed and will be modified.